### DISCUSSION TITLE
[Bug]: Adding the custom command --jetstream for NATS not possible

### DISCUSSION THREAD

**Discussion by SpuriousGer** [coollabsio/coolify#5821] [Feature Requests]
### Description

It's not possible to use custom run flags with Docker images using Coolify at the moment with the exception of very specific ones. This makes Coolify a very limited approach for such use cases.

### Minimal Reproduction (if possible, example repository)

1. Create a new service Docker based
2. use image `nats` with tage `latest`
3. Custom Docker Options: `--jetstream`
4. Using the NATS CLI run `nats server check jetstream --context nats001`

### Exception or Error

`CRITICAL JetStream Crit:JetStream not available: context deadline exceeded`

### Version

v4.0.0-beta.335

### Cloud?

- [ ] Yes
- [X] No

**Comment by SpuriousGer** [comment]
![image](https://github.com/user-attachments/assets/3272ee0b-d533-4c01-8615-f9eccdb4a1fd)
Here is a screenshot of my current config.

**Comment by amerkestijn** [comment]
We have a workaround by mounting (storage) to the nats-server.conf file and loading in our own config.
The custom docker option is not working and I don't know why (yet).

**Comment by sansavision** [comment]
you could try docker compose and run the commands that way.
Here is a small example of the docker compose and the server.conf file. 

docker-compose.yml
services:
  na_1:
    image: nats:latest
    ports:
      - "4222:4222"
    command: "-c /container/path/to/server.conf --name na_1 -p 4222"
    volumes:
      - ./local/path/to/server.conf:/container/path/to/server.conf 
      
      
then in the server.conf file you create the jetstream
server.conf  

jetstream {}

**Comment by skriems** [comment]
I did the same as mentioned by @sansavision.

1. created a new service with a docker compose file
```
  nats1:
    image: nats
    command: -c /etc/nats.server.conf --name na_1 -p 4222
    volumes:
      - ./nats.server.conf:/etc/nats.server.conf
    ports:
      - 4222
      - 9222
```

2. In the `Storages` section turn that mount into a file add your config, i.e.

```
websocket {
  listen: 0.0.0.0:9222
  no_tls: true
  compression: true
}

jetstream {}
```

3. I use caddy and for the websocket I had to go to my dynamic proxy settings on the server page and add a new file like `nats.caddy` with the content

```
nats.example.com {
  @websockets {
      header Connection *Upgrade*
      header Upgrade websocket
  }
  reverse_proxy @websockets <nats-container>:9222
  reverse_proxy <nats-container>:4222
}
```

after that you should be able to use "wss://nats.example.com" like
```bash
nats pub -s "wss://nats.example.com" hello world
```

**Comment by SpuriousGer** [comment]
For whatever reason, I cannot access the nats server with the nats CLI using the chosen custom domain, only via IP and port. Any reason for that?

Plus: What's the benefit of using websockets?

**Comment by Cinzya** [comment]
This is not a bug, but sadly a limitation of Coolify at the moment. The supported Commands are listed in the Docs here: https://coolify.io/docs/knowledge-base/docker/custom-commands
I'll move this thread into the Discussions, so discourse about this feature can continue there.

**Comment by SpuriousGer** [comment]
Is there a reason why such custom commands are so strictly limited? Given the fact that the command is itself only applicable to one application.
